### ISSUE TITLE
[Role API] When Update end point is called with an already existing role name, the update happens instead of returning 'already exists' message

### ISSUE THREAD

**Issue by sdgun** [topcoder-platform/taas-apis#291]
**Steps**
Call update end point for a role by changing the role name to another existing role name

**Current behavior**
Update happens

**Expected behavior**
Should not allow duplicate role names(create end point has this restriction)

**Comment by eisbilir** [comment]
@sdgun 
I found the updated record:

![Postman 2021-06-03 01 03 26](https://user-images.githubusercontent.com/72204071/120558397-0eae0480-c408-11eb-9b33-ced2b283957a.png)

The first "React Developer 2" role actually not exists in database.
I explained it here: https://github.com/topcoder-platform/taas-apis/issues/290#issuecomment-853426363
Now after the above update, we actually have a "React Developer 2" Role.
When I try to update another Role's name to "React Developer 2" I get the error now:
![Postman 2021-06-03 01 05 00](https://user-images.githubusercontent.com/72204071/120558823-b4fa0a00-c408-11eb-94bf-34d1734d8809.png)

@nikolay83 
The undeleted record issue #290 caused this particular issue.
Our update endpoint follow the same `not allow duplicate name` restriction with create endpoint.
And it's not possible to create a new role with duplicate name because `Role` model has a uniqe name index.
